charts/sampling/templates/_agent_config_statefulset.river.txt

```diff
@@ -0,0 +1,17 @@
+{{- define "agent.config.statefulset" -}}
+  {{- include "statefulset.receiver.otlp" . }}
+  {{- if .Values.metricsGeneration.enabled -}}
+    {{- include "statefulset.connector.spanmetrics" . }}
```

Comment:
I included one more processor to the config in docs `otelcol.processor.transform "drop_unneeded_resource_attributes"`: https://github.com/grafana/app-o11y/blob/main/docs/sources/setup/sampling/assets/agent-config-head-sampling.river#L96-L118

It is about dropping unneeded resource attributes in traces before passing them to spanmetrics connector.
That was feedback from Paulin, also I found we do the same for Tempo metrics generation: https://github.com/grafana/app-observability-plugin/blob/671336f758aaf3c0c7a1006c8c4d95baafcc749a/src/constants/overrides.ts#L11

We probably should include this processor in this helm chart as well

Reply:
thanks @mar4uk, nice catch. I'll add this processor here as well.